static/js/settings_ui.js

```diff
@@ -66,6 +66,64 @@ exports.do_settings_change = function (request_method, url, data, status_element
     });
 };
 
+// Generic function for informing users about changes to the settings
+// UI. Uses the save_discard_button_widget used in other parts of the
+// application. Works by hiding the discard_button so that only the
+// save button is visible, and whose text is modified to display the
+// current status.
+exports.do_settings_change_with_button = function (request_method, url, data,
+                                                   status_element, opts) {
+    let success_msg;
+    let success_continuation;
+    let error_continuation;
+
+    const save_button_save_state = settings_org.change_save_button_state;
+    const save_button_div = $(status_element).find(".save-button-controls");
+    const discard_button = $(status_element).find(".discard-button");
+    const save_btn = $(status_element).find(".save-button");
+    const textEl = save_btn.find('.icon-button-text');
+    discard_button.addClass("hide");
+    save_button_div.removeClass('hide').addClass('show').fadeIn(300);
+    save_button_save_state(save_button_div, "saving");
+
+    if (opts !== undefined) {
+        success_msg = opts.success_msg;
+        success_continuation = opts.success_continuation;
+        error_continuation = opts.error_continuation;
+    }
+    if (success_msg === undefined) {
+        success_msg = exports.strings.success;
+    }
+
+    request_method({
+        url: url,
+        data: data,
+        success: function (reponse_data) {
+            save_button_save_state(save_button_div, "succeeded");
+            if (opts !== undefined && opts.success_msg) {
+                textEl.text(opts.success_msg).fadeIn(0);
+            }
+            if (success_continuation !== undefined) {
+                if (opts !== undefined && opts.success_continuation_arg) {
+                    success_continuation(opts.success_continuation_arg);
+                } else {
+                    success_continuation(reponse_data);
+                }
+            }
+        },
+        error: function (xhr) {
+            save_button_save_state(save_button_div, "failed");
+            save_button_div.addClass("hide");
+            if (opts !== undefined && opts.error_msg_element) {
+                ui_report.error(exports.strings.failure, xhr, opts.error_msg_element);
+            }
+            if (error_continuation !== undefined) {
+                error_continuation(xhr);
+            }
+        },
+    });
+};
```

Comment:
It looks like most of this code is copied from `do_settings_change`; what do think about just adding a new `opts` option to `do_settings_change` to support this use case?  It seems like that would be a lot cleaner and more maintainable, since once we duplicate this code, we need to remember to update both copies when making changes.

Reply:
Once all the settings-pages have been changed, the `do_settings_change` function would be redundant and can be deleted.
Since, these were the changes only for the account-settings page. The other pages still use the do_settings_change function, so I kept the function for compatibility with the other pages and created a new function to use for changed pages. Should I still merge the 2 functions?